ambry-router/src/main/java/com.github.ambry.router/DeleteManager.java

```diff
@@ -100,10 +100,11 @@ void submitDeleteBlobOperation(String blobIdString, FutureResult<Void> futureRes
     try {
       BlobId blobId = RouterUtils.getBlobIdFromString(blobIdString, clusterMap);
       DeleteOperation deleteOperation =
-          new DeleteOperation(routerConfig, responseHandler, blobId, futureResult, callback, time);
+          new DeleteOperation(routerConfig, responseHandler, blobId, futureResult, callback, time, routerMetrics);
       deleteOperations.add(deleteOperation);
     } catch (RouterException e) {
       routerMetrics.operationDequeuingRate.mark();
+      routerMetrics.countError(e, NonBlockingRouter.RouterOperationType.DeleteBlob);
```

Comment:
@pnarayanan Re your question
> "avoids potential mistakes when updating metrics for each operation" - how so?

My initial reason is that, for example, at completion of a specific operation, we do not need to look for the exact metric name, but use a general method that is used across different operations. This will help simply remember `countErr()` rather than `xxxBlob[Info]aaa`, where every part needs to be correct. This, however, is a one-time work when producing the code.

I have gone throng the comments you made below. I agree with most of your points, and I will make changes accordingly.